jetstream/core/orchestrator.py

```diff
@@ -714,7 +724,34 @@ def _detokenize_thread(self, idx: int):
       if data is None:
         break
       start_detokenize_time = time.time()
-      if isinstance(data[1], engine_api.ResultTokens):
+      # prefill first token
+      if isinstance(data[0], engine_api.ResultTokens):
```

Comment:
Could we merge the prefill first token block into the generate step tokens block? Since the detokenization logic is identical.

Reply:
The logic are actually different. The generate step are processing all the slots and the available slots are depending on which slots are put into the detokenize_backlog. If we do the same thing for prefill, both prefill and generate threads are putting slot information and tokens into detokenize log, this will make the logic much more complicated. I think leave prefill as separate block making the code easier to understand.